browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -291,6 +291,19 @@ void OnBeforeURLRequestAdBlockTP(const ResponseCallback& next_callback,
       ctx->browser_context && !ctx->browser_context->IsTor() &&
       ProxySettingsAllowUncloaking(ctx->browser_context);
 
+  // When default 1p blocking is disabled, first-party requests should not be
+  // CNAME uncloaked unless using aggressive blocking mode.
+  if (!base::FeatureList::IsEnabled(
+          brave_shields::features::kBraveAdblockDefault1pBlocking) &&
+      should_check_uncloaked && !ctx->aggressive_blocking &&
+      SameDomainOrHost(
+          ctx->request_url,
+          url::Origin::CreateFromNormalizedTuple("https",
```

Comment:
`ctx->initiator_url.GetOrigin()` ?

Reply:
I wanted to use the same logic here as we do in the rest of the adblock engines. We should investigate whether or not that change is appropriate in the other places too - although it'll probably happen with the larger restructuring ahead.